swap_meet/vendor.py

```diff
@@ -1,2 +1,88 @@
+from unicodedata import category
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self, inventory = None):
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory 
+            
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+
+    def get_by_category(self, category):
+        if not category:
+            return False
+        category_list = []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list 
+
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item not in self.inventory:
+            return False
+        if their_item not in friend.inventory:
+            return False
+
+        self.inventory.remove(my_item)
+        friend.inventory.append(my_item) 
+        friend.inventory.remove(their_item)
+        self.inventory.append(their_item)
+
+        if their_item not in self.inventory:
+            return False
+        else:
+            return True
+
+
+    def swap_first_item(self, friend):
+        if self.inventory == []:
+            return False
+        else: 
+            my_item = self.inventory[0]
+        if friend.inventory == []:
+            return False
+        else:
+            their_item = friend.inventory[0]
+        # use swap_items method
+        self.swap_items(friend, my_item, their_item)
+        return True
+
+
+    def get_best_by_category(self, category):
+        desired_category = []
+
+        for item in self.inventory:
+            if category in item.category:
+                desired_category.append(item)
+        # lambda will sort objects by instance's "condtion" attribute
+                sort_list = sorted(desired_category, key=lambda item: item.condition)
```

Comment:
We should line up comments with the code they refer to. This makes it easier to see at a glance the indentation (and thus various scopes) across a file.
```suggestion
                # lambda will sort objects by instance's "condtion" attribute
                sort_list = sorted(desired_category, key=lambda item: item.condition)
```